File: assessment/ADO_boxplot.py
import argparse
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

def plot_custom_boxplot(ax, data, position, color, width=0.5):
    """Helper function to plot a transparent boxplot with matching point colors"""
    data = data.dropna()
    if len(data) == 0:
        return
        
    # Create the boxplot
    bp = ax.boxplot(
        data, 
        positions=[position], 
        widths=width, 
        patch_artist=True, 
        showfliers=False
    )
    
    # Set colors for all boxplot elements
    for element in ['boxes', 'whiskers', 'caps', 'medians']:
        plt.setp(bp[element], color=color, linewidth=1)
        
    # Make the box face transparent
    for patch in bp['boxes']:
        patch.set_facecolor('none')
        
    # Add jittered points
    x_jitter = np.random.normal(position, 0.05, size=len(data))
    ax.scatter(x_jitter, data, color=color, alpha=0.7, s=5, zorder=3)

def main():
    parser = argparse.ArgumentParser(description="Plot ADO for PTA and MDA (Simulated vs Real)")
    parser.add_argument('--mda-sim', required=True, help="MDA Simulated summary TSV")
    parser.add_argument('--mda-real', required=True, help="MDA Real summary TSV")
    parser.add_argument('--pta-sim', required=True, help="PTA Simulated summary TSV")
    parser.add_argument('--pta-real', required=True, help="PTA Real summary TSV")
    parser.add_argument('--output', default="ADO_boxplot_publication.pdf", help="Output plot file")
    
    args = parser.parse_args()

    # 1. Load data
    try:
        mda_sim = pd.read_csv(args.mda_sim, sep='\t')
        mda_real = pd.read_csv(args.mda_real, sep='\t')
        pta_sim = pd.read_csv(args.pta_sim, sep='\t')
        pta_real = pd.read_csv(args.pta_real, sep='\t')
    except Exception as e:
        print(f"Error reading files: {e}")
        return

    # 2. Filter real MDA and PTA for samples ending with "chr10.sort"
    mda_real = mda_real[mda_real['sample'].str.contains(r'chr10\.sort', na=False)]
    pta_real = pta_real[pta_real['sample'].str.contains(r'chr10\.sort', na=False)]

    # 3. Extract ADO and convert to percentage
    # Handling potential missing files/columns gracefully
    get_ado = lambda df: df['ADO_rate'] * 100 if 'ADO_rate' in df.columns else pd.Series(dtype=float)
    
    ado_mda_sim = get_ado(mda_sim)
    ado_pta_sim = get_ado(pta_sim)
    ado_mda_real = get_ado(mda_real)
    ado_pta_real = get_ado(pta_real)

    # 4. Publication-Ready Plot Aesthetics
    plt.rcParams["font.family"] = "sans-serif"
    plt.rcParams["font.sans-serif"] = ["Arial", "Helvetica", "DejaVu Sans"]
    
    color_pta = "#5e3c99" 
    color_mda = "#e66101" 
    # Create figure with 2 subplots (Simulated and Real)
    fig, (ax1, ax2) = plt.subplots(1, 2, figsize=(8, 5), sharey=True)

    # --- Panel 1: Simulated Data ---
    plot_custom_boxplot(ax1, ado_pta_sim, position=1, color=color_pta)
    plot_custom_boxplot(ax1, ado_mda_sim, position=2, color=color_mda)
    
    ax1.set_title("Simulated Data", fontsize=12)
    ax1.set_xticks([1, 2])
    ax1.set_xticklabels(['PTA', 'MDA'], fontweight='bold')
    ax1.set_ylabel('ADO (%)', fontsize=12)
    ax1.set_xlim(0.5, 2.5)

    # --- Panel 2: Real Data ---
    plot_custom_boxplot(ax2, ado_pta_real, position=1, color=color_pta)
    plot_custom_boxplot(ax2, ado_mda_real, position=2, color=color_mda)

    ax2.set_title("Real Data (chr10.sort)", fontsize=12)
    ax2.set_xticks([1, 2])
    ax2.set_xticklabels(['PTA', 'MDA'])
    ax2.set_xlim(0.5, 2.5)

    # Clean up spines, ticks, and add grid/axis limits
    for ax in [ax1, ax2]:
        ax.spines['top'].set_visible(False)
        ax.spines['right'].set_visible(False)
        ax.tick_params(axis='both', which='major', labelsize=11)
        
        # --- NEW CHANGES: Set exact 0-100 limits and add background grid lines ---
        ax.set_ylim(0, 100)
        ax.grid(axis='y', linestyle=':', color='lightgray', alpha=0.7, zorder=0)

    plt.suptitle('Allelic Dropout (ADO) Comparison', fontweight='bold', fontsize=14, y=1.02)
    plt.tight_layout()
    
    # Save high-resolution figures
    plt.savefig(args.output, dpi=600, bbox_inches='tight')
    png_out = args.output.replace('.pdf', '.png')
    plt.savefig(png_out, dpi=600, bbox_inches='tight')
    plt.close()
    
    print(f"Plots saved to {args.output} and {png_out}")

if __name__ == "__main__":
    main()

File: assessment/ADO_boxplot.R
library(ggplot2)
library(dplyr)
library(readr)
library(ggpubr)

############################################################
# 1. Load data
############################################################

mda_sim  <- read_tsv("../1Esfand/MDA_summary.tsv", show_col_types = FALSE)
mda_real <- read_tsv("../1Esfand/real_MDA_summary.tsv", show_col_types = FALSE)
pta_sim  <- read_tsv("../1Esfand/PTA_initi_summary.tsv", show_col_types = FALSE)
pta_real <- read_tsv("../1Esfand/real_PTA_summary.tsv", show_col_types = FALSE)

# Filter real MDA and PTA for samples ending with "chr10.sort"
mda_real <- mda_real %>%
  filter(grepl("chr10\\.sort$", sample))

pta_real <- pta_real %>%
  filter(grepl("chr10\\.sort$", sample))


############################################################
# 2. Add metadata
############################################################

mda_sim  <- mda_sim  %>% mutate(Method = "MDA", Type = "Simulated")
mda_real <- mda_real %>% mutate(Method = "MDA", Type = "Real")

pta_sim  <- pta_sim  %>% mutate(Method = "PTA", Type = "Simulated")
pta_real <- pta_real %>% mutate(Method = "PTA", Type = "Real")

table(mda_real$Method, mda_real$Type)  # Should only show filtered samples
table(pta_real$Method, pta_real$Type)
############################################################
# 3. Merge all
############################################################

df <- bind_rows(mda_sim, mda_real, pta_sim, pta_real)

############################################################
# 4. Convert ADO to percentage
############################################################

df <- df %>%
  mutate(ADO_percent = ADO_rate * 100)

############################################################
# 5. Factor ordering (important for plot layout)
############################################################

df$Method <- factor(df$Method, levels = c("MDA", "PTA"))
df$Type   <- factor(df$Type, levels = c("Simulated", "Real"))

############################################################
# 6. Plot
############################################################
comparisons <- list(
  c("Simulated", "Real")
)

df_means <- df %>%
  group_by(Method, Type) %>%
  summarize(mean_ADO = mean(ADO_percent), .groups = "drop")

p <- ggplot(df, aes(x = Method, y = ADO_percent, fill = Type)) +
  
  # Boxplot
  geom_boxplot(
    width = 0.6,
    position = position_dodge(width = 0.7),
    alpha = 0.9,
    outlier.shape = NA   # hide default outliers (we show raw points)
  ) +
  
  # Jittered points
  geom_jitter(
    position = position_jitterdodge(
      jitter.width = 0.15,
      dodge.width  = 0.7
    ),
    color = "black",
    size = 2,
    alpha = 0.6
  ) +
  
  # Show mean as text on top of boxes
  geom_text(
    data = df_means,
    aes(x = Method, y = mean_ADO + 2, label = sprintf("%.1f", mean_ADO), group = Type),
    position = position_dodge(width = 0.7),
    color = "black",
    size = 4,
    fontface = "bold"
  ) +
  
  # Fill colors
  scale_fill_manual(values = c("Simulated" = "#4C72B0", "Real" = "#DD8452")) +
  
  # Labels
  labs(
    title = "Allelic Dropout (ADO) Comparison",
    subtitle = "MDA vs PTA in Simulated and Real Data (chr10.sort only)",
    x = "Amplification Method",
    y = "ADO (%)"
  ) +
  
  # Theme
  theme_classic(base_size = 14) +
  theme(
    plot.title      = element_text(face = "bold", size = 16),
    plot.subtitle   = element_text(size = 13),
    axis.title      = element_text(face = "bold"),
    axis.text       = element_text(color = "black"),
    legend.title    = element_blank(),
    legend.position = "top"
  ) 
#+
#  
#  stat_compare_means(
#    method = "wilcox.test",
#   aes(group = Type),
#    label = "p.signif"
#  )

############################################################
# 7. Save high-resolution figure
############################################################

ggsave(
  "ADO_boxplot_publication.png",
  p,
  width = 6,
  height = 5,
  dpi = 600
)

ggsave(
  "ADO_boxplot_publication.pdf",
  p,
  width = 6,
  height = 5
)

print(p)